aws-distro-opentelemetry-node-autoinstrumentation/test/exporter/otlp/aws/logs/aws-cw-otlp-batch-log-record-processor.test.ts

```diff
@@ -0,0 +1,244 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import * as sinon from 'sinon';
+import { AnyValue, SeverityNumber, LogRecord } from '@opentelemetry/api-logs';
+import {
+  AwsCloudWatchOtlpBatchLogRecordProcessor,
+  BASE_LOG_BUFFER_BYTE_SIZE,
+  MAX_LOG_REQUEST_BYTE_SIZE,
+} from '../../../../../src/exporter/otlp/aws/logs/aws-cw-otlp-batch-log-record-processor';
+import { OTLPAwsLogExporter } from '../../../../../src/exporter/otlp/aws/logs/otlp-aws-log-exporter';
+import expect from 'expect';
+import { ExportResultCode } from '@opentelemetry/core';
+
+describe('AwsCloudWatchOtlpBatchLogRecordProcessor', () => {
+  describe('estimateLogSize', () => {
+    it('should handle nested structures (dict/array)', () => {
+      const logBody = 'X'.repeat(400);
+      const logKey = 'test';
+      const logDepth = 2;
+
+      const nestedDictLog = generateTestLogData(logBody, logKey, logDepth, 1, true)[0];
+      const nestedArrayLog = generateTestLogData(logBody, logKey, logDepth, 1, false)[0];
+
+      const expectedDictSize = logKey.length * logDepth + logBody.length;
+      const expectedArraySize = logBody.length;
```

Comment:
Isn't logBody just a string of 400 "X"s? I suppose you actually want to create an Array of 400 "X"s instead.